rtos/Mutex.h

```diff
@@ -84,18 +84,10 @@ class Mutex : private mbed::NonCopyable<Mutex> {
     /**
       Wait until a Mutex becomes available.
 
-      @return  status code that indicates the execution status of the function:
-               @a osOK the mutex has been obtained.
-
       @note You cannot call this function from ISR context.
-      @note This function treats RTOS errors as fatal system errors, so it can only return osOK.
-            Use of the return value is deprecated, as the return is expected to become void in the future.
+      @note This function treats RTOS errors as fatal system errors, so it might not return.
```

Comment:
"It might not return" sounds a bit odd. I think I'd be tempted to just stop at the comma, or maybe drop the note altogether.

It was only really to explain that people shouldn't be looking at the return code, and I don't think it's needed any more.